tests/components/watergate/test_init.py

```diff
@@ -0,0 +1,81 @@
+"""Tests for the Watergate integration init module."""
+
+from unittest.mock import patch
+
+from homeassistant.components import webhook
+from homeassistant.components.watergate.const import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import CONF_WEBHOOK_ID
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from .const import MOCK_CONFIG
+
+from tests.common import ANY
+from tests.typing import ClientSessionGenerator
+
+
+async def test_async_setup_entry(
+    hass: HomeAssistant, mock_entry, mock_watergate_client
+) -> None:
+    """Test setting up the Watergate integration."""
+    hass.config.components.add("network")
+    hass.config.internal_url = "http://hassio.local"
```

Comment:
I don't think you need that either (not sure though)

Reply:
It is only to have a little bit prettier assertions instead of some 10.10.10.10 in line 
```
mock_watergate_client.async_set_webhook_url.assert_called_once_with(
            "http://hassio.local/api/webhook/webhook_id"
        )
```